{
 "cells": [
  {
   "cell_type": "markdown",
   "source": [
    "# Homology Modelling of Side Chains"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "aa916824f567455e"
  },
  {
   "cell_type": "code",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Homology Modelling: GFP/FM_10. Min Homology = 0.35. Maximum Number of Templates to Use = 5.\n"
     ]
    }
   ],
   "source": [
    "from pathlib import Path\n",
    "\n",
    "# change accordingly\n",
    "GENERATED_CLS = GENERATED_CLS if \"GENERATED_CLS\" in locals() else \"GFP\"\n",
    "GENERATED_SID = GENERATED_SID if \"GENERATED_SID\" in locals() else \"FM_10\"\n",
    "assert GENERATED_CLS is not None\n",
    "assert GENERATED_SID is not None\n",
    "\n",
    "MIN_HOMOLOGY = MIN_HOMOLOGY if \"MIN_HOMOLOGY\" in locals() else 0.35\n",
    "MIN_TM_SCORE = MIN_TM_SCORE if \"MIN_TM_SCORE\" in locals() else 0.8\n",
    "MAX_TEMPLATES = MAX_TEMPLATES if \"MAX_TEMPLATES\" in locals() else 5\n",
    "print(\n",
    "    f\"Homology Modelling: {GENERATED_CLS}/{GENERATED_SID}. Min Homology = {MIN_HOMOLOGY}. Maximum Number of Templates to Use = {MAX_TEMPLATES}.\")\n",
    "\n",
    "GENERATED_PDB = Path(f\"../../data/{GENERATED_CLS}/generated/BQ/{GENERATED_SID}.pdb\")\n",
    "GENERATED_SEQ = Path(f\"../../data/{GENERATED_CLS}/generated/Q/{GENERATED_SID}.fasta\")"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2025-03-25T08:23:27.585277Z",
     "start_time": "2025-03-25T08:23:27.582085Z"
    }
   },
   "id": "2f5be2ae8ee4ca3f",
   "execution_count": 1
  },
  {
   "cell_type": "markdown",
   "source": [
    "Identy and select template structures for subsequent homology modelling of side chains.\n",
    "\n",
    "- Use FoldSeek for sequence + structure alignment.\n",
    "- The homology should be at least 70%.\n",
    "\n",
    "You may wish to increase the limit on the data rate that the notebook server can send to clients and the window size used to compute the average data rate for the limit, by adding the following tags: \n",
    "\n",
    "```\n",
    "notebook --ServerApp.iopub_data_rate_limit 10000000.0 --ServerApp.rate_limit_window=25.0\n",
    "```"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "fcd40ebc116671f9"
  },
  {
   "cell_type": "code",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job submitted: {'id': 'SC7dw7VFfKF09_WuI__16gQYtB_NEa6qdo1z_A', 'status': 'COMPLETE'}\n",
      "Running status: {'id': 'SC7dw7VFfKF09_WuI__16gQYtB_NEa6qdo1z_A', 'status': 'COMPLETE'}\n"
     ]
    }
   ],
   "source": [
    "from time import sleep\n",
    "\n",
    "from requests import get, post\n",
    "\n",
    "TICKET_URL = \"https://search.foldseek.com/api/ticket\"\n",
    "RESULT_URL = \"https://search.foldseek.com/api/result\"\n",
    "\n",
    "# submit a new job\n",
    "# curl -X POST -F q=@{GEN_PDB} -F 'mode=tmalign' -F 'database[]=pdb100' https://search.foldseek.com/api/ticket\n",
    "with open(GENERATED_PDB, 'rb') as f:\n",
    "    ticket = post(\n",
    "        TICKET_URL,\n",
    "        files={'q': f},\n",
    "        data={\n",
    "            \"mode\": \"tmalign\",\n",
    "            \"database[]\": \"pdb100\"\n",
    "        }).json()\n",
    "    print(\"Job submitted:\", ticket)\n",
    "\n",
    "repeat = True\n",
    "while repeat:\n",
    "    status = get(f\"{TICKET_URL}/{ticket['id']}\").json()\n",
    "    print(\"Running status:\", ticket)\n",
    "    if status[\"status\"] == \"ERROR\":\n",
    "        break\n",
    "    sleep(1)\n",
    "    repeat = status[\"status\"] != \"COMPLETE\"\n",
    "\n",
    "result = get(f\"{RESULT_URL}/{ticket['id']}/0\").json()\n",
    "# import json\n",
    "# with open(\"result.json\", \"w\") as json_file:\n",
    "#     json.dump(result, json_file, indent=4)\n",
    "result = result[\"results\"][0][\"alignments\"][0]\n",
    "\n",
    "visited = set()\n",
    "templates = [\n",
    "                {\"PDB\": pdb,\n",
    "                 \"seq\": alignment[\"tSeq\"],\n",
    "                 \"tm-score\": alignment[\"eval\"],\n",
    "                 \"prob\": alignment[\"prob\"],\n",
    "                 \"seq-per-identity\": alignment[\"seqId\"],\n",
    "                 }\n",
    "                for alignment in result\n",
    "                if alignment[\"seqId\"] > MIN_HOMOLOGY\n",
    "                if alignment[\"eval\"] > MIN_HOMOLOGY\n",
    "                if (pdb := alignment[\"target\"].split(\"-\")[0].upper()) not in visited and not visited.add(pdb)\n",
    "                if pdb not in [\"6WJM\", \"6WGP\", \"2QU1\"]\n",
    "            ][:min(MAX_TEMPLATES, len(result))]"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2025-03-25T08:23:36.215501Z",
     "start_time": "2025-03-25T08:23:30.881846Z"
    }
   },
   "id": "a2f6ff83f1e3dc61",
   "execution_count": 2
  },
  {
   "cell_type": "markdown",
   "source": [
    "Prepare (1) working directory, (2) FASTAs for alignment, (3) PDBs for homology modelling."
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "3cd54ae3d9f18b6a"
  },
  {
   "cell_type": "code",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Template 1Q4D_A\n",
      "Template 5NI3_D\n",
      "Template 5NI3_B\n",
      "Template 5NI3_C\n",
      "Template 5NI3_A\n",
      "Template 1EMA_A\n",
      "Template 2DUI_A\n",
      "Template 1YFP_A\n",
      "Template 1YFP_B\n"
     ]
    }
   ],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio import PDB, SeqIO\n",
    "import shutil\n",
    "\n",
    "wd = Path(f\"wd/{GENERATED_CLS}/{GENERATED_SID}\")\n",
    "wd.mkdir(parents=True, exist_ok=True)\n",
    "\n",
    "pdb_dir = wd / \"pdb\"\n",
    "pdb_dir.mkdir(parents=True, exist_ok=True)\n",
    "\n",
    "fasta = wd / \"seqs.fasta\"\n",
    "fasta_seqs = [SeqIO.read(GENERATED_SEQ, \"fasta\")]\n",
    "parser = PDB.PDBParser()\n",
    "\n",
    "shutil.copy(\n",
    "    src=GENERATED_PDB,\n",
    "    dst=pdb_dir / GENERATED_PDB.name\n",
    ")\n",
    "\n",
    "for temp in templates:\n",
    "    temp_id = temp[\"PDB\"]\n",
    "    for temp_pdb in Path(f\"../../data/{GENERATED_CLS}/processed\").glob(f\"{temp_id}*.pdb\"):\n",
    "        print(f\"Template {temp_pdb.stem}\")\n",
    "        shutil.copy(\n",
    "            src=temp_pdb,\n",
    "            dst=pdb_dir / temp_pdb.name\n",
    "        )\n",
    "        structure = parser.get_structure('structure', temp_pdb)\n",
    "        sequence = ''\n",
    "        for model in structure:\n",
    "            for chain in model:\n",
    "                for residue in chain:\n",
    "                    if PDB.is_aa(residue, standard=True):\n",
    "                        sequence += PDB.Polypeptide.index_to_one(PDB.Polypeptide.three_to_index(residue.resname))\n",
    "        fasta_seqs.append(SeqRecord(Seq(sequence), id=temp_pdb.stem, description=\"\"))\n",
    "\n",
    "with open(fasta, \"w\") as f:\n",
    "    SeqIO.write(fasta_seqs, f, format=\"fasta\")"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2025-03-25T08:23:36.406790Z",
     "start_time": "2025-03-25T08:23:36.216637Z"
    }
   },
   "id": "6547ece576a5a16b",
   "execution_count": 3
  },
  {
   "cell_type": "markdown",
   "source": [
    "Align sequences using Clustal."
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "90c2815fe1bf47f"
  },
  {
   "cell_type": "code",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "clustalo -i wd/GFP/FM_10/seqs.fasta -o wd/GFP/FM_10/seqs.aln.fasta --outfmt=fa\r\n",
      "seqret -sequence wd/GFP/FM_10/seqs.aln.fasta -outseq wd/GFP/FM_10/seqs.pir -osformat2 pir\r\n",
      "Read and write (return) sequences\r\n"
     ]
    }
   ],
   "source": [
    "# !clustalw -infile={fasta} -output=pir\n",
    "\n",
    "!echo clustalo -i {fasta} -o {fasta.parent}/{fasta.stem}.aln.fasta --outfmt=fa\n",
    "!clustalo -i {fasta} -o {fasta.parent}/{fasta.stem}.aln.fasta --outfmt=fa\n",
    "\n",
    "!echo seqret -sequence {fasta.parent}/{fasta.stem}.aln.fasta -outseq {fasta.parent}/{fasta.stem}.pir -osformat2 pir\n",
    "!seqret -sequence {fasta.parent}/{fasta.stem}.aln.fasta -outseq {fasta.parent}/{fasta.stem}.pir -osformat2 pir"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2025-03-25T08:24:01.491974Z",
     "start_time": "2025-03-25T08:24:00.924697Z"
    }
   },
   "id": "502f1d824bc4a506",
   "execution_count": 4
  },
  {
   "cell_type": "markdown",
   "source": [
    "Fill the missing details in the PIR alignment."
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "cffd1cce415d4db5"
  },
  {
   "cell_type": "code",
   "outputs": [],
   "source": [
    "PIR_RAW = fasta.parent / f\"{fasta.stem}.pir\"\n",
    "PIR_REF = PIR_RAW.parent / f\"{PIR_RAW.stem}.ref.pir\"\n",
    "\n",
    "entries = []\n",
    "\n",
    "# l1: >P1;8DOD\n",
    "# l2: StructureX:8DOD.pdb:26:A:290:::::\n",
    "# lx: <SEQUENCE>\n",
    "with open(PIR_RAW, \"r\") as i, open(PIR_REF, \"w\") as o:\n",
    "    entry_id = None\n",
    "    l2 = False\n",
    "    for l in i:\n",
    "        # l1\n",
    "        if l.startswith(\">P1;\"):\n",
    "            entry_id = l[4:].split(\" \")[0].strip()\n",
    "            l2 = True\n",
    "            o.write(l)\n",
    "        # l2\n",
    "        elif l2:\n",
    "            l2 = False\n",
    "            chain_id = None\n",
    "            try:\n",
    "                chain_id = next(Path(f\"../../data/{GENERATED_CLS}/processed\").glob(f\"{entry_id}*.pdb\")).stem.split(\"_\")[-1]\n",
    "            except StopIteration:\n",
    "                chain_id = \"A\"\n",
    "            start, end = float('inf'), float('-inf')\n",
    "            for model in parser.get_structure('structure', pdb_dir / f\"{entry_id}.pdb\"):\n",
    "                for chain in model:\n",
    "                    if chain.id != chain_id:\n",
    "                        continue\n",
    "                    for residue in chain:\n",
    "                        if PDB.is_aa(residue, standard=True):\n",
    "                            start = min(residue.id[1], start)\n",
    "                            end = max(residue.id[1], start)\n",
    "            o.write(f\"StructureX:{entry_id}.pdb:{start}:{chain_id}:{end}:::::\\n\")\n",
    "            entries.append(entry_id)\n",
    "        # lx\n",
    "        else:\n",
    "            o.write(l)\n",
    "\n",
    "entries.remove(GENERATED_SID)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2025-03-25T08:24:02.617444Z",
     "start_time": "2025-03-25T08:24:02.543269Z"
    }
   },
   "id": "ddb5c25d4217cefa",
   "execution_count": 5
  },
  {
   "cell_type": "markdown",
   "source": [
    "Homology modelling of side chains.\n",
    "Main chains are fixed.\n",
    "Load model from the existing backbone structure.\n",
    "\n",
    "Model 10 times; select the best one.\n",
    "\n",
    "You may need to manually edit the data to avoid potential errors."
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "22836faf1eef550c"
  },
  {
   "cell_type": "code",
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/tz365/miniconda3/envs/proteval/lib/python3.10/site-packages/IPython/core/magics/osm.py:417: UserWarning: This is now an optional IPython functionality, setting dhist requires you to install the `pickleshare` library.\n",
      "  self.shell.db['dhist'] = compress_dhist(dhist)[-100:]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/tz365/WS/PROTEVAL/notebooks/side-chain-modelng/wd/GFP/FM_10\n",
      "\n",
      "                         MODELLER 10.6, 2024/10/17, r12888\n",
      "\n",
      "     PROTEIN STRUCTURE MODELLING BY SATISFACTION OF SPATIAL RESTRAINTS\n",
      "\n",
      "\n",
      "                     Copyright(c) 1989-2024 Andrej Sali\n",
      "                            All Rights Reserved\n",
      "\n",
      "                             Written by A. Sali\n",
      "                               with help from\n",
      "              B. Webb, M.S. Madhusudhan, M-Y. Shen, G.Q. Dong,\n",
      "          M.A. Marti-Renom, N. Eswar, F. Alber, M. Topf, B. Oliva,\n",
      "             A. Fiser, R. Sanchez, B. Yerkovich, A. Badretdinov,\n",
      "                     F. Melo, J.P. Overington, E. Feyfant\n",
      "                 University of California, San Francisco, USA\n",
      "                    Rockefeller University, New York, USA\n",
      "                      Harvard University, Cambridge, USA\n",
      "                   Imperial Cancer Research Fund, London, UK\n",
      "              Birkbeck College, University of London, London, UK\n",
      "\n",
      "\n",
      "Kind, OS, HostName, Kernel, Processor: 4, Linux 23OCT-4090-70K 5.15.167.4-microsoft-standard-WSL2 x86_64\n",
      "Date and time of compilation         : 2024/10/17 20:34:40\n",
      "MODELLER executable type             : x86_64-intel8\n",
      "Job starting time (YY/MM/DD HH:MM:SS): 2025/03/25 16:24:03\n",
      "\n",
      "IOData_____W> Setting io.atom_files_directory to a colon-delimited string is\n",
      "              deprecated, as it is not robust on Windows systems. Set it to\n",
      "              a list of directories instead. For example:\n",
      "              env.io.atom_files_directory = ['./pdb']\n",
      "readlinef__W> File: ./pdb/1Q4D_A.pdb, Line: 1792\n",
      "              Modeller will only read the first 80 characters of this line.\n",
      "\n",
      "readlinef__W> File: ./pdb/5NI3_D.pdb, Line: 1795\n",
      "              Modeller will only read the first 80 characters of this line.\n",
      "\n",
      "readlinef__W> File: ./pdb/5NI3_B.pdb, Line: 1819\n",
      "              Modeller will only read the first 80 characters of this line.\n",
      "\n",
      "readlinef__W> File: ./pdb/5NI3_C.pdb, Line: 1814\n",
      "              Modeller will only read the first 80 characters of this line.\n",
      "\n",
      "readlinef__W> File: ./pdb/5NI3_A.pdb, Line: 1790\n",
      "              Modeller will only read the first 80 characters of this line.\n",
      "\n",
      "readlinef__W> File: ./pdb/2DUI_A.pdb, Line: 1777\n",
      "              Modeller will only read the first 80 characters of this line.\n",
      "\n",
      "readlinef__W> File: ./pdb/1YFP_A.pdb, Line: 1678\n",
      "              Modeller will only read the first 80 characters of this line.\n",
      "\n",
      "readlinef__W> File: ./pdb/1YFP_B.pdb, Line: 1678\n",
      "              Modeller will only read the first 80 characters of this line.\n",
      "\n",
      "\n",
      "check_ali___> Checking pairwise structural superpositions. \n",
      "\n",
      "Equivalent CA pairs with distance difference larger than  6.0 angstroms:\n",
      "\n",
      "ALN_POS TMPL1 TMPL2  RID1  RID2  NAM1  NAM2     DIST\n",
      "----------------------------------------------------\n",
      "    232     3     4  233   233      M     M    8.311\n",
      "END OF TABLE\n",
      "\n",
      "check_ali___> Checking the sequence-structure alignment. \n",
      "\n",
      "Implied intrachain target CA(i)-CA(i+1) distances longer than  8.0 angstroms:\n",
      "\n",
      "ALN_POS  TMPL  RID1  RID2  NAM1  NAM2     DIST\n",
      "----------------------------------------------\n",
      "END OF TABLE\n",
      "read_to_681_> topology.submodel read from topology file:        3\n",
      "readlinef__W> File: ./pdb/FM_10.pdb, Line: 1141\n",
      "              Modeller will only read the first 80 characters of this line.\n",
      "\n",
      "mdtrsr__446W> A potential that relies on one protein is used, yet you have at\n",
      "              least one known structure available. MDT, not library, potential is used.\n",
      "0 atoms in HETATM/BLK residues constrained\n",
      "to protein atoms within 2.30 angstroms\n",
      "and protein CA atoms within 10.00 angstroms\n",
      "0 atoms in residues without defined topology\n",
      "constrained to be rigid bodies\n",
      "condens_443_> Restraints marked for deletion were removed.\n",
      "              Total number of restraints before, now:    21284    19636\n",
      "800 (of 1713 total) atoms selected for optimization\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  LEU:OXT  LEU\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      228\n",
      "Number of all, selected real atoms                :     1713     800\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    19636    7115\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   231820\n",
      "Dynamic pairs routine                             : 5, NATM_SEL x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :      -13591.4756\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -13591.475586\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      228\n",
      "Number of all, selected real atoms                :     1713     800\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    19636    7115\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     2296\n",
      "Dynamic pairs routine                             : 5, NATM_SEL x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :        3740.7593\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :     841       0      1   0.007   0.007      10.739       1.000\n",
      " 2 Bond angle potential               :    1240       1      6   2.542   2.542      152.56       1.000\n",
      " 3 Stereochemical cosine torsion poten:    1081       0     38  52.228  52.228      466.99       1.000\n",
      " 4 Stereochemical improper torsion pot:     449       1      4   2.345   2.345      42.443       1.000\n",
      " 5 Soft-sphere overlap restraints     :    2296       0      1   0.007   0.007      13.482       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "10 Distance restraints 2 (N-O)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     183       0      5  79.880  79.880      80.690       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:     134       0      4  75.593  75.593      53.490       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:      65       0      0  57.869  57.869      31.287       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      28       0      0 103.042 103.042      20.785       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :    2433       0      0   0.629   0.629      1971.9       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :     661      15     64   1.989   1.989      896.36       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "# Heavy relative violation of each residue is written to: FM_10.V99990001\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   14571.5869\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature  4                           : Stereochemical improper torsion potentia\n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1   5438  68F  68F CA  N     495  494   56.36   35.27   21.10    5.03   35.27   21.10    5.03\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 26                           : Distance restraints 4 (SDCH-SDCH)       \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1  19002   7L  83C CB  SG     45  604   12.33    4.80    7.53    4.96    4.80    7.53    4.96\n",
      "    2  19004  11V  68F CB  CD1    76  498   11.31    5.11    6.20    5.15    5.11    6.20    5.15\n",
      "    3  19005  11V  68F CB  CE1    76  500   10.38    3.80    6.58    5.85    3.80    6.58    5.85\n",
      "    4  19006  11V  68F CB  CZ     76  502   10.56    3.90    6.66    5.89    3.90    6.66    5.89\n",
      "    5  19008  11V  68F CG1 CE1    77  500   10.24    4.37    5.87    5.05    4.37    5.87    5.05\n",
      "    6  19009  11V  68F CG1 CZ     77  502   10.30    3.97    6.33    5.55    3.97    6.33    5.55\n",
      "    7  19017  11V  68F CG2 CE1    78  500   10.17    4.23    5.94    5.10    4.23    5.94    5.10\n",
      "    8  19018  11V  68F CG2 CZ     78  502   10.54    4.67    5.87    4.99    4.67    5.87    4.99\n",
      "    9  19033  13V  68F CB  CE2    90  501   10.96    5.08    5.88    4.87    5.08    5.88    4.87\n",
      "   10  19182  42L  68F CG  CB    304  496   10.83    4.17    6.67    5.38    4.17    6.67    5.38\n",
      "   11  19194  42L  68F CD2 CB    306  496    9.98    3.80    6.18    5.08    3.80    6.18    5.08\n",
      "   12  19254  66Y  82C CB  CB    478  597   13.33    5.00    8.33    6.57    5.00    8.33    6.57\n",
      "   13  19255  66Y  82C CB  SG    478  598   12.21    4.17    8.04    6.73    4.17    8.04    6.73\n",
      "   14  19259  67S  83C CB  SG    490  604   14.33    4.37    9.97    8.50    4.37    9.97    8.50\n",
      "   15  19319  82C 159I SG  CD1   598 1170   11.27    5.07    6.21    4.55    5.07    6.21    4.55\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "  serious non-bonded atom clash:    88  235   2.241\n",
      "  serious non-bonded atom clash:   246  300   2.028\n",
      "  serious non-bonded atom clash:   263  286   1.945\n",
      "  serious non-bonded atom clash:   444  454   2.237\n",
      "  serious non-bonded atom clash:   444  456   2.060\n",
      "  serious non-bonded atom clash:   616  638   2.107\n",
      "  serious non-bonded atom clash:  1037 1041   1.968\n",
      "  serious non-bonded atom clash:  1228 1235   2.208\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     4    1    3    2    4   12   31   91  116  168  182  181  193  212  287\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  LEU:OXT  LEU\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      228\n",
      "Number of all, selected real atoms                :     1713     800\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    19636    7115\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   228956\n",
      "Dynamic pairs routine                             : 5, NATM_SEL x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :      -13257.1289\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -13257.128906\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      228\n",
      "Number of all, selected real atoms                :     1713     800\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    19636    7115\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     2210\n",
      "Dynamic pairs routine                             : 5, NATM_SEL x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :        3691.7717\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :     841       0      1   0.009   0.009      16.570       1.000\n",
      " 2 Bond angle potential               :    1240       1      7   2.599   2.599      157.31       1.000\n",
      " 3 Stereochemical cosine torsion poten:    1081       0     39  52.087  52.087      463.84       1.000\n",
      " 4 Stereochemical improper torsion pot:     449       1      2   2.212   2.212      38.057       1.000\n",
      " 5 Soft-sphere overlap restraints     :    2210       0      2   0.009   0.009      22.986       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "10 Distance restraints 2 (N-O)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     183       0      8  77.825  77.825      81.332       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:     134       0      0  69.852  69.852      33.222       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:      65       0      0  61.668  61.668      37.978       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      28       0      0  81.551  81.551      17.287       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :    2433       0      3   0.564   0.564      1941.1       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :     661      15     72   1.909   1.909      882.10       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "# Heavy relative violation of each residue is written to: FM_10.V99990002\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   13412.6562\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature  4                           : Stereochemical improper torsion potentia\n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1   5438  68F  68F CA  N     495  494   57.49   35.27   22.23    5.30   35.27   22.23    5.30\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 26                           : Distance restraints 4 (SDCH-SDCH)       \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1  19002   7L  83C CB  SG     45  604   11.99    4.80    7.19    4.73    4.80    7.19    4.73\n",
      "    2  19004  11V  68F CB  CD1    76  498   11.27    5.11    6.16    5.12    5.11    6.16    5.12\n",
      "    3  19005  11V  68F CB  CE1    76  500   10.31    3.80    6.51    5.79    3.80    6.51    5.79\n",
      "    4  19006  11V  68F CB  CZ     76  502   10.41    3.90    6.51    5.75    3.90    6.51    5.75\n",
      "    5  19008  11V  68F CG1 CE1    77  500   10.17    4.37    5.80    4.99    4.37    5.80    4.99\n",
      "    6  19009  11V  68F CG1 CZ     77  502   10.14    3.97    6.17    5.41    3.97    6.17    5.41\n",
      "    7  19017  11V  68F CG2 CE1    78  500   10.08    4.23    5.84    5.02    4.23    5.84    5.02\n",
      "    8  19018  11V  68F CG2 CZ     78  502   10.37    4.67    5.71    4.84    4.67    5.71    4.84\n",
      "    9  19033  13V  68F CB  CE2    90  501   10.81    5.08    5.74    4.75    5.08    5.74    4.75\n",
      "   10  19182  42L  68F CG  CB    304  496   10.81    4.17    6.64    5.35    4.17    6.64    5.35\n",
      "   11  19194  42L  68F CD2 CB    306  496    9.96    3.80    6.16    5.06    3.80    6.16    5.06\n",
      "   12  19254  66Y  82C CB  CB    478  597   13.62    5.00    8.62    6.79    5.00    8.62    6.79\n",
      "   13  19255  66Y  82C CB  SG    478  598   12.24    4.17    8.08    6.76    4.17    8.08    6.76\n",
      "   14  19259  67S  83C CB  SG    490  604   12.63    4.37    8.27    7.05    4.37    8.27    7.05\n",
      "   15  19315  82C  90Y CB  OH    597  651   10.28    4.20    6.08    4.90    4.20    6.08    4.90\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "  serious non-bonded atom clash:    88  235   2.241\n",
      "  serious non-bonded atom clash:   246  300   2.028\n",
      "  serious non-bonded atom clash:   263  286   1.945\n",
      "  serious non-bonded atom clash:   444  454   2.237\n",
      "  serious non-bonded atom clash:   444  456   2.060\n",
      "  serious non-bonded atom clash:   616  638   2.107\n",
      "  serious non-bonded atom clash:  1037 1041   1.968\n",
      "  serious non-bonded atom clash:  1228 1235   2.208\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     4    1    3    2    4   16   32   93  109  167  173  174  199  219  271\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  LEU:OXT  LEU\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      228\n",
      "Number of all, selected real atoms                :     1713     800\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    19636    7115\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   231041\n",
      "Dynamic pairs routine                             : 5, NATM_SEL x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :      -12955.1777\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -12955.177734\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      228\n",
      "Number of all, selected real atoms                :     1713     800\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    19636    7115\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     2307\n",
      "Dynamic pairs routine                             : 5, NATM_SEL x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :        3841.2319\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :     841       0      1   0.009   0.009      16.454       1.000\n",
      " 2 Bond angle potential               :    1240       1      5   2.845   2.845      188.02       1.000\n",
      " 3 Stereochemical cosine torsion poten:    1081       0     40  52.187  52.187      469.44       1.000\n",
      " 4 Stereochemical improper torsion pot:     449       2      6   2.873   2.873      62.872       1.000\n",
      " 5 Soft-sphere overlap restraints     :    2307       0      0   0.006   0.006      11.098       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "10 Distance restraints 2 (N-O)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     183       0      9  81.408  81.408      122.16       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:     134       0      5  76.683  76.683      59.859       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:      65       0      0  58.943  58.943      28.979       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      28       0      0  71.162  71.162      17.674       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :    2433       0      2   0.630   0.630      1972.6       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :     661      14     72   1.957   1.957      892.12       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "# Heavy relative violation of each residue is written to: FM_10.V99990003\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   14607.1299\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature  4                           : Stereochemical improper torsion potentia\n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1   5438  68F  68F CA  N     495  494   59.56   35.27   24.30    5.79   35.27   24.30    5.79\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 26                           : Distance restraints 4 (SDCH-SDCH)       \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1  19004  11V  68F CB  CD1    76  498   11.13    5.11    6.02    5.00    5.11    6.02    5.00\n",
      "    2  19005  11V  68F CB  CE1    76  500   10.14    3.80    6.34    5.64    3.80    6.34    5.64\n",
      "    3  19006  11V  68F CB  CZ     76  502   10.28    3.90    6.38    5.64    3.90    6.38    5.64\n",
      "    4  19008  11V  68F CG1 CE1    77  500    9.97    4.37    5.61    4.82    4.37    5.61    4.82\n",
      "    5  19009  11V  68F CG1 CZ     77  502    9.98    3.97    6.01    5.27    3.97    6.01    5.27\n",
      "    6  19017  11V  68F CG2 CE1    78  500    9.97    4.23    5.73    4.93    4.23    5.73    4.93\n",
      "    7  19018  11V  68F CG2 CZ     78  502   10.31    4.67    5.64    4.79    4.67    5.64    4.79\n",
      "    8  19033  13V  68F CB  CE2    90  501   10.64    5.08    5.56    4.61    5.08    5.56    4.61\n",
      "    9  19182  42L  68F CG  CB    304  496   10.75    4.17    6.59    5.31    4.17    6.59    5.31\n",
      "   10  19194  42L  68F CD2 CB    306  496    9.92    3.80    6.12    5.02    3.80    6.12    5.02\n",
      "   11  19254  66Y  82C CB  CB    478  597   13.46    5.00    8.46    6.67    5.00    8.46    6.67\n",
      "   12  19255  66Y  82C CB  SG    478  598   14.02    4.17    9.85    8.24    4.17    9.85    8.24\n",
      "   13  19259  67S  83C CB  SG    490  604   12.37    4.37    8.00    6.82    4.37    8.00    6.82\n",
      "   14  19319  82C 159I SG  CD1   598 1170   11.67    5.07    6.60    4.83    5.07    6.60    4.83\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "  serious non-bonded atom clash:    88  235   2.241\n",
      "  serious non-bonded atom clash:   246  300   2.028\n",
      "  serious non-bonded atom clash:   263  286   1.945\n",
      "  serious non-bonded atom clash:   444  454   2.237\n",
      "  serious non-bonded atom clash:   444  456   2.060\n",
      "  serious non-bonded atom clash:   616  638   2.107\n",
      "  serious non-bonded atom clash:  1037 1041   1.968\n",
      "  serious non-bonded atom clash:  1228 1235   2.208\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     4    1    3    2    4   14   30   92  117  152  186  198  210  228  281\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_484W> Dihedral still outside +-90:      -90.4302\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  LEU:OXT  LEU\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      228\n",
      "Number of all, selected real atoms                :     1713     800\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    19636    7115\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   231021\n",
      "Dynamic pairs routine                             : 5, NATM_SEL x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :      -13733.8203\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -13733.820312\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      228\n",
      "Number of all, selected real atoms                :     1713     800\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    19636    7115\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     2216\n",
      "Dynamic pairs routine                             : 5, NATM_SEL x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :        3722.9419\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :     841       1      3   0.010   0.010      19.756       1.000\n",
      " 2 Bond angle potential               :    1240       2      3   2.646   2.646      166.98       1.000\n",
      " 3 Stereochemical cosine torsion poten:    1081       0     42  52.267  52.267      465.72       1.000\n",
      " 4 Stereochemical improper torsion pot:     449       1      2   2.113   2.113      34.075       1.000\n",
      " 5 Soft-sphere overlap restraints     :    2216       1      5   0.012   0.012      34.606       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "10 Distance restraints 2 (N-O)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     183       0      9  77.525  77.525      92.325       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:     134       0      1  69.883  69.883      38.931       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:      65       0      0  64.608  64.608      36.314       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      28       0      0  76.748  76.748      15.796       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :    2433       0      1   0.570   0.570      1949.7       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :     661      15     62   1.857   1.857      868.74       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "# Heavy relative violation of each residue is written to: FM_10.V99990004\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   13424.7812\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature  1                           : Bond length potential                   \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1   1576 206I 206I CD1 CG1  1541 1539    1.69    1.53    0.17    4.58    1.53    0.17    4.58\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature  4                           : Stereochemical improper torsion potentia\n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1   5438  68F  68F CA  N     495  494   57.08   35.27   21.82    5.20   35.27   21.82    5.20\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 26                           : Distance restraints 4 (SDCH-SDCH)       \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1  19002   7L  83C CB  SG     45  604   12.28    4.80    7.48    4.92    4.80    7.48    4.92\n",
      "    2  19004  11V  68F CB  CD1    76  498   11.31    5.11    6.20    5.15    5.11    6.20    5.15\n",
      "    3  19005  11V  68F CB  CE1    76  500   10.35    3.80    6.55    5.82    3.80    6.55    5.82\n",
      "    4  19006  11V  68F CB  CZ     76  502   10.47    3.90    6.57    5.81    3.90    6.57    5.81\n",
      "    5  19008  11V  68F CG1 CE1    77  500   10.19    4.37    5.82    5.01    4.37    5.82    5.01\n",
      "    6  19009  11V  68F CG1 CZ     77  502   10.48    3.97    6.52    5.72    3.97    6.52    5.72\n",
      "    7  19033  13V  68F CB  CE2    90  501   10.83    5.08    5.75    4.76    5.08    5.75    4.76\n",
      "    8  19182  42L  68F CG  CB    304  496   10.80    4.17    6.63    5.35    4.17    6.63    5.35\n",
      "    9  19194  42L  68F CD2 CB    306  496   10.18    3.80    6.38    5.24    3.80    6.38    5.24\n",
      "   10  19195  42L  68F CD2 CG    306  497    9.51    3.80    5.71    4.64    3.80    5.71    4.64\n",
      "   11  19197  42L  68F CD2 CD2   306  499    9.29    3.73    5.56    4.61    3.73    5.56    4.61\n",
      "   12  19254  66Y  82C CB  CB    478  597   13.41    5.00    8.41    6.63    5.00    8.41    6.63\n",
      "   13  19255  66Y  82C CB  SG    478  598   12.01    4.17    7.84    6.57    4.17    7.84    6.57\n",
      "   14  19259  67S  83C CB  SG    490  604   14.34    4.37    9.97    8.50    4.37    9.97    8.50\n",
      "   15  19315  82C  90Y CB  OH    597  651   10.09    4.20    5.89    4.75    4.20    5.89    4.75\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "  serious non-bonded atom clash:    88  235   2.241\n",
      "  serious non-bonded atom clash:   246  300   2.028\n",
      "  serious non-bonded atom clash:   263  286   1.945\n",
      "  serious non-bonded atom clash:   444  454   2.237\n",
      "  serious non-bonded atom clash:   444  456   2.060\n",
      "  serious non-bonded atom clash:   616  638   2.107\n",
      "  serious non-bonded atom clash:  1037 1041   1.968\n",
      "  serious non-bonded atom clash:  1228 1235   2.208\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     4    1    3    2    5   12   28  101   96  153  167  173  190  223  274\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  LEU:OXT  LEU\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      228\n",
      "Number of all, selected real atoms                :     1713     800\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    19636    7115\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   229857\n",
      "Dynamic pairs routine                             : 5, NATM_SEL x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :      -13266.0605\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -13266.060547\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      228\n",
      "Number of all, selected real atoms                :     1713     800\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    19636    7115\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     2267\n",
      "Dynamic pairs routine                             : 5, NATM_SEL x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :        3728.2651\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :     841       0      2   0.007   0.007      10.212       1.000\n",
      " 2 Bond angle potential               :    1240       3      6   2.565   2.565      153.27       1.000\n",
      " 3 Stereochemical cosine torsion poten:    1081       0     39  52.384  52.384      465.49       1.000\n",
      " 4 Stereochemical improper torsion pot:     449       0      0   1.800   1.800      24.957       1.000\n",
      " 5 Soft-sphere overlap restraints     :    2267       0      1   0.008   0.008      16.809       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "10 Distance restraints 2 (N-O)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     183       0      7  77.397  77.397      87.728       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:     134       0      1  73.981  73.981      44.280       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:      65       0      0  63.817  63.817      36.989       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      28       0      0  83.432  83.432      17.990       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :    2433       0      3   0.575   0.575      1959.3       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :     661      18     68   2.013   2.013      911.26       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "# Heavy relative violation of each residue is written to: FM_10.V99990005\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   13751.9951\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature  2                           : Bond angle potential                    \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1   3741 193K 193K C   CA   1433 1427  131.49  108.00   23.49    5.44  108.00   23.49    5.44\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 26                           : Distance restraints 4 (SDCH-SDCH)       \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1  19004  11V  68F CB  CD1    76  498   12.27    5.11    7.16    5.95    5.11    7.16    5.95\n",
      "    2  19005  11V  68F CB  CE1    76  500   11.77    3.80    7.97    7.08    3.80    7.97    7.08\n",
      "    3  19006  11V  68F CB  CZ     76  502   12.18    3.90    8.28    7.32    3.90    8.28    7.32\n",
      "    4  19008  11V  68F CG1 CE1    77  500   11.56    4.37    7.19    6.18    4.37    7.19    6.18\n",
      "    5  19009  11V  68F CG1 CZ     77  502   11.91    3.97    7.95    6.97    3.97    7.95    6.97\n",
      "    6  19017  11V  68F CG2 CE1    78  500   11.66    4.23    7.43    6.39    4.23    7.43    6.39\n",
      "    7  19018  11V  68F CG2 CZ     78  502   12.23    4.67    7.56    6.42    4.67    7.56    6.42\n",
      "    8  19033  13V  68F CB  CE2    90  501   11.69    5.08    6.62    5.48    5.08    6.62    5.48\n",
      "    9  19174  36A  68F CB  CD1   266  498    9.49    4.30    5.19    4.61    4.30    5.19    4.61\n",
      "   10  19182  42L  68F CG  CB    304  496   10.90    4.17    6.73    5.43    4.17    6.73    5.43\n",
      "   11  19194  42L  68F CD2 CB    306  496   10.56    3.80    6.76    5.55    3.80    6.76    5.55\n",
      "   12  19195  42L  68F CD2 CG    306  497    9.86    3.80    6.06    4.93    3.80    6.06    4.93\n",
      "   13  19197  42L  68F CD2 CD2   306  499    9.63    3.73    5.89    4.89    3.73    5.89    4.89\n",
      "   14  19254  66Y  82C CB  CB    478  597   13.47    5.00    8.47    6.68    5.00    8.47    6.68\n",
      "   15  19255  66Y  82C CB  SG    478  598   14.01    4.17    9.84    8.24    4.17    9.84    8.24\n",
      "   16  19259  67S  83C CB  SG    490  604   12.67    4.37    8.30    7.08    4.37    8.30    7.08\n",
      "   17  19274  68F 117L CE2 CB    501  869   10.06    4.17    5.89    4.59    4.17    5.89    4.59\n",
      "   18  19319  82C 159I SG  CD1   598 1170   11.69    5.07    6.63    4.86    5.07    6.63    4.86\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "  serious non-bonded atom clash:    88  235   2.241\n",
      "  serious non-bonded atom clash:   246  300   2.028\n",
      "  serious non-bonded atom clash:   263  286   1.945\n",
      "  serious non-bonded atom clash:   444  454   2.237\n",
      "  serious non-bonded atom clash:   444  456   2.060\n",
      "  serious non-bonded atom clash:   616  638   2.107\n",
      "  serious non-bonded atom clash:  1037 1041   1.968\n",
      "  serious non-bonded atom clash:  1228 1235   2.208\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     4    1    3    2    4   13   29  106  113  165  193  164  197  216  269\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_484W> Dihedral still outside +-90:      -90.9157\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  LEU:OXT  LEU\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      228\n",
      "Number of all, selected real atoms                :     1713     800\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    19636    7115\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   231933\n",
      "Dynamic pairs routine                             : 5, NATM_SEL x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :      -13931.1719\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -13931.171875\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      228\n",
      "Number of all, selected real atoms                :     1713     800\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    19636    7115\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     2107\n",
      "Dynamic pairs routine                             : 5, NATM_SEL x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :        3714.2766\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :     841       0      1   0.007   0.007      9.7158       1.000\n",
      " 2 Bond angle potential               :    1240       1      7   2.707   2.707      170.88       1.000\n",
      " 3 Stereochemical cosine torsion poten:    1081       0     43  52.269  52.269      465.50       1.000\n",
      " 4 Stereochemical improper torsion pot:     449       1      2   2.176   2.176      36.512       1.000\n",
      " 5 Soft-sphere overlap restraints     :    2107       0      0   0.005   0.005      6.7396       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "10 Distance restraints 2 (N-O)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     183       0      8  73.808  73.808      88.847       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:     134       0      3  71.353  71.353      49.804       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:      65       0      0  64.769  64.769      38.368       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      28       0      0  79.131  79.131      20.885       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :    2433       0      1   0.560   0.560      1944.1       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :     661      16     65   1.959   1.959      882.95       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "# Heavy relative violation of each residue is written to: FM_10.V99990006\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   13431.3955\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature  4                           : Stereochemical improper torsion potentia\n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1   5438  68F  68F CA  N     495  494   55.73   35.27   20.46    4.88   35.27   20.46    4.88\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 26                           : Distance restraints 4 (SDCH-SDCH)       \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1  19002   7L  83C CB  SG     45  604   12.24    4.80    7.44    4.89    4.80    7.44    4.89\n",
      "    2  19004  11V  68F CB  CD1    76  498   11.32    5.11    6.21    5.16    5.11    6.21    5.16\n",
      "    3  19005  11V  68F CB  CE1    76  500   10.43    3.80    6.63    5.89    3.80    6.63    5.89\n",
      "    4  19006  11V  68F CB  CZ     76  502   10.69    3.90    6.79    6.00    3.90    6.79    6.00\n",
      "    5  19008  11V  68F CG1 CE1    77  500   10.29    4.37    5.92    5.09    4.37    5.92    5.09\n",
      "    6  19009  11V  68F CG1 CZ     77  502   10.42    3.97    6.46    5.66    3.97    6.46    5.66\n",
      "    7  19017  11V  68F CG2 CE1    78  500   10.20    4.23    5.97    5.13    4.23    5.97    5.13\n",
      "    8  19018  11V  68F CG2 CZ     78  502   10.64    4.67    5.98    5.07    4.67    5.98    5.07\n",
      "    9  19033  13V  68F CB  CE2    90  501   11.05    5.08    5.97    4.94    5.08    5.97    4.94\n",
      "   10  19182  42L  68F CG  CB    304  496   10.82    4.17    6.65    5.36    4.17    6.65    5.36\n",
      "   11  19194  42L  68F CD2 CB    306  496   10.05    3.80    6.25    5.13    3.80    6.25    5.13\n",
      "   12  19254  66Y  82C CB  CB    478  597   13.58    5.00    8.58    6.76    5.00    8.58    6.76\n",
      "   13  19255  66Y  82C CB  SG    478  598   12.45    4.17    8.29    6.94    4.17    8.29    6.94\n",
      "   14  19256  66Y  90Y CB  OH    478  651   10.08    4.43    5.65    4.59    4.43    5.65    4.59\n",
      "   15  19259  67S  83C CB  SG    490  604   14.31    4.37    9.94    8.48    4.37    9.94    8.48\n",
      "   16  19319  82C 159I SG  CD1   598 1170   12.10    5.07    7.03    5.15    5.07    7.03    5.15\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "  serious non-bonded atom clash:    88  235   2.241\n",
      "  serious non-bonded atom clash:   246  300   2.028\n",
      "  serious non-bonded atom clash:   263  286   1.945\n",
      "  serious non-bonded atom clash:   444  454   2.237\n",
      "  serious non-bonded atom clash:   444  456   2.060\n",
      "  serious non-bonded atom clash:   616  638   2.107\n",
      "  serious non-bonded atom clash:  1037 1041   1.968\n",
      "  serious non-bonded atom clash:  1228 1235   2.208\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     4    1    3    2    4   10   30   85  101  146  161  165  195  226  269\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  LEU:OXT  LEU\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      228\n",
      "Number of all, selected real atoms                :     1713     800\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    19636    7115\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   230696\n",
      "Dynamic pairs routine                             : 5, NATM_SEL x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :      -13356.2832\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -13356.283203\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      228\n",
      "Number of all, selected real atoms                :     1713     800\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    19636    7115\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     2242\n",
      "Dynamic pairs routine                             : 5, NATM_SEL x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :        3805.0056\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :     841       1      2   0.010   0.010      19.315       1.000\n",
      " 2 Bond angle potential               :    1240       3      9   3.142   3.142      227.54       1.000\n",
      " 3 Stereochemical cosine torsion poten:    1081       0     41  52.099  52.099      465.55       1.000\n",
      " 4 Stereochemical improper torsion pot:     449       1      3   2.347   2.347      42.336       1.000\n",
      " 5 Soft-sphere overlap restraints     :    2242       1      4   0.011   0.011      31.662       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "10 Distance restraints 2 (N-O)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     183       0      8  83.009  83.009      92.511       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:     134       0      3  76.433  76.433      47.369       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:      65       0      0  61.855  61.855      37.801       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      28       0      0  76.496  76.496      18.121       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :    2433       0      1   0.572   0.572      1963.2       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :     661      14     62   1.815   1.815      859.58       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "# Heavy relative violation of each residue is written to: FM_10.V99990007\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   14639.3457\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature  1                           : Bond length potential                   \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1   1461 193K 193K CG  CB   1429 1428    1.70    1.53    0.17    4.56    1.53    0.17    4.56\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature  2                           : Bond angle potential                    \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1   3733 193K 193K N   CA   1426 1427  150.11  113.50   36.61    9.85  113.50   36.61    9.85\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature  4                           : Stereochemical improper torsion potentia\n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1   5438  68F  68F CA  N     495  494   56.71   35.27   21.44    5.11   35.27   21.44    5.11\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 26                           : Distance restraints 4 (SDCH-SDCH)       \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1  19004  11V  68F CB  CD1    76  498   11.28    5.11    6.17    5.13    5.11    6.17    5.13\n",
      "    2  19005  11V  68F CB  CE1    76  500   10.39    3.80    6.59    5.85    3.80    6.59    5.85\n",
      "    3  19006  11V  68F CB  CZ     76  502   10.64    3.90    6.74    5.96    3.90    6.74    5.96\n",
      "    4  19008  11V  68F CG1 CE1    77  500   10.20    4.37    5.83    5.02    4.37    5.83    5.02\n",
      "    5  19009  11V  68F CG1 CZ     77  502   10.33    3.97    6.37    5.59    3.97    6.37    5.59\n",
      "    6  19017  11V  68F CG2 CE1    78  500   10.19    4.23    5.96    5.12    4.23    5.96    5.12\n",
      "    7  19018  11V  68F CG2 CZ     78  502   10.64    4.67    5.97    5.07    4.67    5.97    5.07\n",
      "    8  19033  13V  68F CB  CE2    90  501   10.89    5.08    5.82    4.82    5.08    5.82    4.82\n",
      "    9  19182  42L  68F CG  CB    304  496   10.84    4.17    6.67    5.38    4.17    6.67    5.38\n",
      "   10  19194  42L  68F CD2 CB    306  496    9.99    3.80    6.19    5.09    3.80    6.19    5.09\n",
      "   11  19254  66Y  82C CB  CB    478  597   13.49    5.00    8.49    6.69    5.00    8.49    6.69\n",
      "   12  19255  66Y  82C CB  SG    478  598   12.42    4.17    8.25    6.91    4.17    8.25    6.91\n",
      "   13  19259  67S  83C CB  SG    490  604   12.43    4.37    8.06    6.87    4.37    8.06    6.87\n",
      "   14  19319  82C 159I SG  CD1   598 1170   11.34    5.07    6.27    4.59    5.07    6.27    4.59\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "  serious non-bonded atom clash:    88  235   2.241\n",
      "  serious non-bonded atom clash:   246  300   2.028\n",
      "  serious non-bonded atom clash:   263  286   1.945\n",
      "  serious non-bonded atom clash:   444  454   2.237\n",
      "  serious non-bonded atom clash:   444  456   2.060\n",
      "  serious non-bonded atom clash:   616  638   2.107\n",
      "  serious non-bonded atom clash:  1037 1041   1.968\n",
      "  serious non-bonded atom clash:  1228 1235   2.208\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     4    1    3    2    5   17   32   99  112  169  171  186  193  239  284\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  LEU:OXT  LEU\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      228\n",
      "Number of all, selected real atoms                :     1713     800\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    19636    7115\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   230173\n",
      "Dynamic pairs routine                             : 5, NATM_SEL x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :      -13319.3994\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -13319.399414\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      228\n",
      "Number of all, selected real atoms                :     1713     800\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    19636    7115\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     2236\n",
      "Dynamic pairs routine                             : 5, NATM_SEL x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :        3755.0142\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :     841       0      1   0.009   0.009      15.966       1.000\n",
      " 2 Bond angle potential               :    1240       1      3   2.564   2.564      154.00       1.000\n",
      " 3 Stereochemical cosine torsion poten:    1081       0     38  52.253  52.253      463.15       1.000\n",
      " 4 Stereochemical improper torsion pot:     449       1      2   2.141   2.141      35.674       1.000\n",
      " 5 Soft-sphere overlap restraints     :    2236       0      2   0.009   0.009      22.804       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "10 Distance restraints 2 (N-O)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     183       0      5  83.295  83.295      92.916       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:     134       0      0  71.946  71.946      37.839       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:      65       0      1  50.592  50.592      31.430       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      28       0      0  80.015  80.015      15.252       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :    2433       0      1   0.615   0.615      1962.2       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :     661      15     76   2.070   2.070      923.76       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "# Heavy relative violation of each residue is written to: FM_10.V99990008\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   14083.1465\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature  4                           : Stereochemical improper torsion potentia\n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1   5438  68F  68F CA  N     495  494   56.01   35.27   20.75    4.95   35.27   20.75    4.95\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 26                           : Distance restraints 4 (SDCH-SDCH)       \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1  19002   7L  83C CB  SG     45  604   12.26    4.80    7.46    4.91    4.80    7.46    4.91\n",
      "    2  19004  11V  68F CB  CD1    76  498   11.33    5.11    6.22    5.16    5.11    6.22    5.16\n",
      "    3  19005  11V  68F CB  CE1    76  500   10.44    3.80    6.64    5.90    3.80    6.64    5.90\n",
      "    4  19006  11V  68F CB  CZ     76  502   10.66    3.90    6.76    5.98    3.90    6.76    5.98\n",
      "    5  19008  11V  68F CG1 CE1    77  500   10.29    4.37    5.92    5.09    4.37    5.92    5.09\n",
      "    6  19009  11V  68F CG1 CZ     77  502   10.39    3.97    6.43    5.64    3.97    6.43    5.64\n",
      "    7  19017  11V  68F CG2 CE1    78  500   10.21    4.23    5.98    5.14    4.23    5.98    5.14\n",
      "    8  19018  11V  68F CG2 CZ     78  502   10.63    4.67    5.96    5.06    4.67    5.96    5.06\n",
      "    9  19033  13V  68F CB  CE2    90  501   11.03    5.08    5.95    4.93    5.08    5.95    4.93\n",
      "   10  19182  42L  68F CG  CB    304  496   10.82    4.17    6.66    5.37    4.17    6.66    5.37\n",
      "   11  19194  42L  68F CD2 CB    306  496   10.01    3.80    6.21    5.10    3.80    6.21    5.10\n",
      "   12  19254  66Y  82C CB  CB    478  597   13.55    5.00    8.55    6.74    5.00    8.55    6.74\n",
      "   13  19255  66Y  82C CB  SG    478  598   12.22    4.17    8.05    6.74    4.17    8.05    6.74\n",
      "   14  19259  67S  83C CB  SG    490  604   14.34    4.37    9.97    8.50    4.37    9.97    8.50\n",
      "   15  19315  82C  90Y CB  OH    597  651   10.18    4.20    5.98    4.82    4.20    5.98    4.82\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "  serious non-bonded atom clash:    88  235   2.241\n",
      "  serious non-bonded atom clash:   246  300   2.028\n",
      "  serious non-bonded atom clash:   263  286   1.945\n",
      "  serious non-bonded atom clash:   444  454   2.237\n",
      "  serious non-bonded atom clash:   444  456   2.060\n",
      "  serious non-bonded atom clash:   616  638   2.107\n",
      "  serious non-bonded atom clash:  1037 1041   1.968\n",
      "  serious non-bonded atom clash:  1228 1235   2.208\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     4    1    3    2    4   15   31  102   92  168  159  181  195  240  282\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  LEU:OXT  LEU\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      228\n",
      "Number of all, selected real atoms                :     1713     800\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    19636    7115\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   231766\n",
      "Dynamic pairs routine                             : 5, NATM_SEL x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :      -13616.0430\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -13616.042969\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      228\n",
      "Number of all, selected real atoms                :     1713     800\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    19636    7115\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     2220\n",
      "Dynamic pairs routine                             : 5, NATM_SEL x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :        3727.8965\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :     841       0      1   0.007   0.007      9.3837       1.000\n",
      " 2 Bond angle potential               :    1240       1      5   2.450   2.450      141.55       1.000\n",
      " 3 Stereochemical cosine torsion poten:    1081       0     38  52.069  52.069      463.87       1.000\n",
      " 4 Stereochemical improper torsion pot:     449       0      4   2.131   2.131      34.434       1.000\n",
      " 5 Soft-sphere overlap restraints     :    2220       0      1   0.006   0.006      8.5713       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "10 Distance restraints 2 (N-O)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     183       0      6  79.583  79.583      93.956       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:     134       0      4  78.924  78.924      52.916       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:      65       0      0  59.988  59.988      31.280       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      28       0      0  90.905  90.905      18.370       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :    2433       0      8   0.628   0.628      1981.9       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :     661      15     66   1.973   1.973      891.63       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "# Heavy relative violation of each residue is written to: FM_10.V99990009\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   14217.6699\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 26                           : Distance restraints 4 (SDCH-SDCH)       \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1  19002   7L  83C CB  SG     45  604   12.01    4.80    7.21    4.74    4.80    7.21    4.74\n",
      "    2  19004  11V  68F CB  CD1    76  498   11.87    5.11    6.76    5.61    5.11    6.76    5.61\n",
      "    3  19005  11V  68F CB  CE1    76  500   11.21    3.80    7.41    6.59    3.80    7.41    6.59\n",
      "    4  19006  11V  68F CB  CZ     76  502   11.53    3.90    7.63    6.74    3.90    7.63    6.74\n",
      "    5  19008  11V  68F CG1 CE1    77  500   11.01    4.37    6.65    5.71    4.37    6.65    5.71\n",
      "    6  19009  11V  68F CG1 CZ     77  502   11.25    3.97    7.28    6.39    3.97    7.28    6.39\n",
      "    7  19017  11V  68F CG2 CE1    78  500   11.10    4.23    6.87    5.90    4.23    6.87    5.90\n",
      "    8  19018  11V  68F CG2 CZ     78  502   11.59    4.67    6.92    5.88    4.67    6.92    5.88\n",
      "    9  19033  13V  68F CB  CE2    90  501   11.28    5.08    6.20    5.14    5.08    6.20    5.14\n",
      "   10  19182  42L  68F CG  CB    304  496   10.90    4.17    6.73    5.43    4.17    6.73    5.43\n",
      "   11  19194  42L  68F CD2 CB    306  496   10.06    3.80    6.26    5.14    3.80    6.26    5.14\n",
      "   12  19254  66Y  82C CB  CB    478  597   13.48    5.00    8.48    6.69    5.00    8.48    6.69\n",
      "   13  19255  66Y  82C CB  SG    478  598   12.43    4.17    8.26    6.92    4.17    8.26    6.92\n",
      "   14  19259  67S  83C CB  SG    490  604   12.30    4.37    7.94    6.77    4.37    7.94    6.77\n",
      "   15  19319  82C 159I SG  CD1   598 1170   12.91    5.07    7.84    5.75    5.07    7.84    5.75\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "  serious non-bonded atom clash:    88  235   2.241\n",
      "  serious non-bonded atom clash:   246  300   2.028\n",
      "  serious non-bonded atom clash:   263  286   1.945\n",
      "  serious non-bonded atom clash:   444  454   2.237\n",
      "  serious non-bonded atom clash:   444  456   2.060\n",
      "  serious non-bonded atom clash:   616  638   2.107\n",
      "  serious non-bonded atom clash:  1037 1041   1.968\n",
      "  serious non-bonded atom clash:  1228 1235   2.208\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     4    1    3    2    4   14   27  100  131  141  158  175  191  233  271\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_484W> Dihedral still outside +-90:       91.0711\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  LEU:OXT  LEU\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      228\n",
      "Number of all, selected real atoms                :     1713     800\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    19636    7115\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   229799\n",
      "Dynamic pairs routine                             : 5, NATM_SEL x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :      -13674.7764\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -13674.776367\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      228\n",
      "Number of all, selected real atoms                :     1713     800\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    19636    7115\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     2091\n",
      "Dynamic pairs routine                             : 5, NATM_SEL x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :        3690.2375\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :     841       0      0   0.006   0.006      8.0557       1.000\n",
      " 2 Bond angle potential               :    1240       1      5   2.379   2.379      132.72       1.000\n",
      " 3 Stereochemical cosine torsion poten:    1081       0     41  52.097  52.097      465.78       1.000\n",
      " 4 Stereochemical improper torsion pot:     449       0      0   1.839   1.839      26.357       1.000\n",
      " 5 Soft-sphere overlap restraints     :    2091       0      0   0.006   0.006      8.9257       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "10 Distance restraints 2 (N-O)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     183       0      5  77.938  77.938      73.567       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:     134       0      2  77.417  77.417      33.215       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:      65       0      0  60.020  60.020      27.728       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      28       0      0  77.187  77.187      17.377       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :    2433       0      4   0.597   0.597      1963.8       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :     661      18     69   2.075   2.075      932.75       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "# Heavy relative violation of each residue is written to: FM_10.V99990010\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   13498.2930\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 26                           : Distance restraints 4 (SDCH-SDCH)       \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1  19002   7L  83C CB  SG     45  604   12.03    4.80    7.23    4.75    4.80    7.23    4.75\n",
      "    2  19004  11V  68F CB  CD1    76  498   14.09    5.11    8.98    7.45    5.11    8.98    7.45\n",
      "    3  19005  11V  68F CB  CE1    76  500   13.93    3.80   10.13    9.00    3.80   10.13    9.00\n",
      "    4  19006  11V  68F CB  CZ     76  502   13.14    3.90    9.24    8.17    3.90    9.24    8.17\n",
      "    5  19008  11V  68F CG1 CE1    77  500   13.84    4.37    9.47    8.14    4.37    9.47    8.14\n",
      "    6  19009  11V  68F CG1 CZ     77  502   12.93    3.97    8.96    7.86    3.97    8.96    7.86\n",
      "    7  19017  11V  68F CG2 CE1    78  500   13.99    4.23    9.76    8.39    4.23    9.76    8.39\n",
      "    8  19018  11V  68F CG2 CZ     78  502   13.24    4.67    8.58    7.28    4.67    8.58    7.28\n",
      "    9  19033  13V  68F CB  CE2    90  501   11.70    5.08    6.62    5.48    5.08    6.62    5.48\n",
      "   10  19174  36A  68F CB  CD1   266  498   11.03    4.30    6.73    5.98    4.30    6.73    5.98\n",
      "   11  19182  42L  68F CG  CB    304  496   10.98    4.17    6.81    5.49    4.17    6.81    5.49\n",
      "   12  19194  42L  68F CD2 CB    306  496   10.66    3.80    6.86    5.64    3.80    6.86    5.64\n",
      "   13  19195  42L  68F CD2 CG    306  497   10.01    3.80    6.21    5.05    3.80    6.21    5.05\n",
      "   14  19197  42L  68F CD2 CD2   306  499    9.82    3.73    6.09    5.05    3.73    6.09    5.05\n",
      "   15  19254  66Y  82C CB  CB    478  597   13.65    5.00    8.65    6.82    5.00    8.65    6.82\n",
      "   16  19255  66Y  82C CB  SG    478  598   12.31    4.17    8.15    6.82    4.17    8.15    6.82\n",
      "   17  19259  67S  83C CB  SG    490  604   12.32    4.37    7.95    6.78    4.37    7.95    6.78\n",
      "   18  19278  68F 117L CZ  CB    502  869    9.95    4.10    5.85    4.56    4.10    5.85    4.56\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "  serious non-bonded atom clash:    88  235   2.241\n",
      "  serious non-bonded atom clash:   246  300   2.028\n",
      "  serious non-bonded atom clash:   263  286   1.945\n",
      "  serious non-bonded atom clash:   444  454   2.237\n",
      "  serious non-bonded atom clash:   444  456   2.060\n",
      "  serious non-bonded atom clash:   616  638   2.107\n",
      "  serious non-bonded atom clash:  1037 1041   1.968\n",
      "  serious non-bonded atom clash:  1228 1235   2.208\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     4    1    3    2    4   14   36   94  105  154  165  164  194  210  250\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "\n",
      ">> Summary of successfully produced models:\n",
      "Filename                          molpdf     DOPE score\n",
      "-------------------------------------------------------\n",
      "FM_10.B99990001.pdb           3740.75928   -13591.47559\n",
      "FM_10.B99990002.pdb           3691.77173   -13257.12891\n",
      "FM_10.B99990003.pdb           3841.23193   -12955.17773\n",
      "FM_10.B99990004.pdb           3722.94189   -13733.82031\n",
      "FM_10.B99990005.pdb           3728.26514   -13266.06055\n",
      "FM_10.B99990006.pdb           3714.27661   -13931.17188\n",
      "FM_10.B99990007.pdb           3805.00562   -13356.28320\n",
      "FM_10.B99990008.pdb           3755.01416   -13319.39941\n",
      "FM_10.B99990009.pdb           3727.89648   -13616.04297\n",
      "FM_10.B99990010.pdb           3690.23755   -13674.77637\n"
     ]
    }
   ],
   "source": [
    "%cd {wd}\n",
    "\n",
    "from modeller import *\n",
    "from modeller.automodel import *\n",
    "from modeller.scripts import *\n",
    "\n",
    "# log.verbose()\n",
    "env = Environ()\n",
    "\n",
    "\n",
    "class SidechainModel(AutoModel):\n",
    "    def select_atoms(self):\n",
    "        return Selection(self).only_sidechain()\n",
    "\n",
    "\n",
    "env.io.atom_files_directory = \"./pdb\"\n",
    "\n",
    "a = SidechainModel(\n",
    "    env,\n",
    "    alnfile=PIR_REF.name,\n",
    "    knowns=entries,\n",
    "    sequence=GENERATED_SID,\n",
    "    inifile=str(f\"./pdb/{GENERATED_PDB.name}\")\n",
    ")\n",
    "a.starting_model = 1\n",
    "a.ending_model = 10\n",
    "a.assess_methods = [assess.DOPE]\n",
    "a.make()"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2025-03-25T08:25:23.130480Z",
     "start_time": "2025-03-25T08:24:03.418361Z"
    }
   },
   "id": "628cf748f71a7c76",
   "execution_count": 6
  },
  {
   "cell_type": "markdown",
   "source": [
    "DOPE is a Z-score.\n",
    "\n",
    "Positive scores are likely to be poor models, while scores lower than -1 or so are likely to be native-like.\n",
    "\n",
    "Select the best model with the lowest score."
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "e085caa392236e6f"
  },
  {
   "cell_type": "code",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "readlinef__W> File: FM_10.B99990008.pdb, Line: 6\n",
      "              Modeller will only read the first 80 characters of this line.\n",
      "\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  LEU:OXT  LEU\n",
      "preppdf_453W> No fixed restraints selected; there may be some dynamic ones.\n",
      "preppdf_454W> Restraints file was probably not read; use restraints.append().\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      228\n",
      "Number of all, selected real atoms                :     1713    1713\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :        0       0\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   326626\n",
      "Dynamic pairs routine                             : 1, NATM x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :      -20044.4883\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -20044.488281\n",
      ">> Normalized DOPE z score: 0.105\n",
      "readlinef__W> File: FM_10.B99990003.pdb, Line: 6\n",
      "              Modeller will only read the first 80 characters of this line.\n",
      "\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  LEU:OXT  LEU\n",
      "preppdf_453W> No fixed restraints selected; there may be some dynamic ones.\n",
      "preppdf_454W> Restraints file was probably not read; use restraints.append().\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      228\n",
      "Number of all, selected real atoms                :     1713    1713\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :        0       0\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   327500\n",
      "Dynamic pairs routine                             : 1, NATM x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :      -19680.0645\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -19680.064453\n",
      ">> Normalized DOPE z score: 0.218\n",
      "readlinef__W> File: FM_10.B99990010.pdb, Line: 6\n",
      "              Modeller will only read the first 80 characters of this line.\n",
      "\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  LEU:OXT  LEU\n",
      "preppdf_453W> No fixed restraints selected; there may be some dynamic ones.\n",
      "preppdf_454W> Restraints file was probably not read; use restraints.append().\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      228\n",
      "Number of all, selected real atoms                :     1713    1713\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :        0       0\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   326258\n",
      "Dynamic pairs routine                             : 1, NATM x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :      -20399.3867\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -20399.386719\n",
      ">> Normalized DOPE z score: -0.005\n",
      "readlinef__W> File: FM_10.B99990006.pdb, Line: 6\n",
      "              Modeller will only read the first 80 characters of this line.\n",
      "\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  LEU:OXT  LEU\n",
      "preppdf_453W> No fixed restraints selected; there may be some dynamic ones.\n",
      "preppdf_454W> Restraints file was probably not read; use restraints.append().\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      228\n",
      "Number of all, selected real atoms                :     1713    1713\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :        0       0\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   328396\n",
      "Dynamic pairs routine                             : 1, NATM x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :      -20655.7637\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -20655.763672\n",
      ">> Normalized DOPE z score: -0.085\n",
      "readlinef__W> File: FM_10.B99990005.pdb, Line: 6\n",
      "              Modeller will only read the first 80 characters of this line.\n",
      "\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  LEU:OXT  LEU\n",
      "preppdf_453W> No fixed restraints selected; there may be some dynamic ones.\n",
      "preppdf_454W> Restraints file was probably not read; use restraints.append().\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      228\n",
      "Number of all, selected real atoms                :     1713    1713\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :        0       0\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   326324\n",
      "Dynamic pairs routine                             : 1, NATM x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :      -19991.2793\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -19991.279297\n",
      ">> Normalized DOPE z score: 0.122\n",
      "readlinef__W> File: FM_10.B99990001.pdb, Line: 6\n",
      "              Modeller will only read the first 80 characters of this line.\n",
      "\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  LEU:OXT  LEU\n",
      "preppdf_453W> No fixed restraints selected; there may be some dynamic ones.\n",
      "preppdf_454W> Restraints file was probably not read; use restraints.append().\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      228\n",
      "Number of all, selected real atoms                :     1713    1713\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :        0       0\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   328277\n",
      "Dynamic pairs routine                             : 1, NATM x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :      -20316.8867\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -20316.886719\n",
      ">> Normalized DOPE z score: 0.021\n",
      "readlinef__W> File: FM_10.B99990007.pdb, Line: 6\n",
      "              Modeller will only read the first 80 characters of this line.\n",
      "\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  LEU:OXT  LEU\n",
      "preppdf_453W> No fixed restraints selected; there may be some dynamic ones.\n",
      "preppdf_454W> Restraints file was probably not read; use restraints.append().\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      228\n",
      "Number of all, selected real atoms                :     1713    1713\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :        0       0\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   327156\n",
      "Dynamic pairs routine                             : 1, NATM x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :      -20081.6328\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -20081.632812\n",
      ">> Normalized DOPE z score: 0.094\n",
      "readlinef__W> File: FM_10.B99990009.pdb, Line: 6\n",
      "              Modeller will only read the first 80 characters of this line.\n",
      "\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  LEU:OXT  LEU\n",
      "preppdf_453W> No fixed restraints selected; there may be some dynamic ones.\n",
      "preppdf_454W> Restraints file was probably not read; use restraints.append().\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      228\n",
      "Number of all, selected real atoms                :     1713    1713\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :        0       0\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   328222\n",
      "Dynamic pairs routine                             : 1, NATM x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :      -20341.1348\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -20341.134766\n",
      ">> Normalized DOPE z score: 0.013\n",
      "readlinef__W> File: FM_10.B99990004.pdb, Line: 6\n",
      "              Modeller will only read the first 80 characters of this line.\n",
      "\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  LEU:OXT  LEU\n",
      "preppdf_453W> No fixed restraints selected; there may be some dynamic ones.\n",
      "preppdf_454W> Restraints file was probably not read; use restraints.append().\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      228\n",
      "Number of all, selected real atoms                :     1713    1713\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :        0       0\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   327481\n",
      "Dynamic pairs routine                             : 1, NATM x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :      -20459.0391\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -20459.039062\n",
      ">> Normalized DOPE z score: -0.024\n",
      "readlinef__W> File: FM_10.B99990002.pdb, Line: 6\n",
      "              Modeller will only read the first 80 characters of this line.\n",
      "\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  LEU:OXT  LEU\n",
      "preppdf_453W> No fixed restraints selected; there may be some dynamic ones.\n",
      "preppdf_454W> Restraints file was probably not read; use restraints.append().\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      228\n",
      "Number of all, selected real atoms                :     1713    1713\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :        0       0\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   325419\n",
      "Dynamic pairs routine                             : 1, NATM x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :      -19982.2363\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -19982.236328\n",
      ">> Normalized DOPE z score: 0.124\n",
      "                 Model  DOPE Score\n",
      "3  FM_10.B99990006.pdb   -0.084752\n",
      "8  FM_10.B99990004.pdb   -0.023645\n",
      "2  FM_10.B99990010.pdb   -0.005116\n",
      "7  FM_10.B99990009.pdb    0.012978\n",
      "5  FM_10.B99990001.pdb    0.020510\n",
      "6  FM_10.B99990007.pdb    0.093585\n",
      "0  FM_10.B99990008.pdb    0.105123\n",
      "4  FM_10.B99990005.pdb    0.121651\n",
      "9  FM_10.B99990002.pdb    0.124460\n",
      "1  FM_10.B99990003.pdb    0.218321\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "\n",
    "lowest_dope = float('inf')\n",
    "best_model = None\n",
    "\n",
    "scores = pd.DataFrame([\n",
    "    {\"Model\": str(model),\n",
    "     \"DOPE Score\": complete_pdb(env, str(model)).assess_normalized_dope()\n",
    "     } for model in Path(\"./\").glob(f\"{GENERATED_SID}.B*.pdb\")\n",
    "]).sort_values(by=\"DOPE Score\", ascending=True)\n",
    "print(scores)\n",
    "\n",
    "os.rename(scores.iloc[0]['Model'], f\"{GENERATED_SID}.best.pdb\")"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2025-03-25T08:25:23.591753Z",
     "start_time": "2025-03-25T08:25:23.131508Z"
    }
   },
   "id": "e06413472133bbc8",
   "execution_count": 7
  },
  {
   "cell_type": "markdown",
   "source": [
    "Verify the modelled side-chains using tools such as:\n",
    "- PROCHECK\n",
    "- Verify3D\n",
    "- WHATCHECK\n",
    "\n",
    "Some are available at https://saves.mbi.ucla.edu/."
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "b42695c83acbc4c8"
  },
  {
   "cell_type": "code",
   "outputs": [],
   "source": [],
   "metadata": {
    "collapsed": false
   },
   "id": "818e413995c47a78",
   "execution_count": null
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
